go.mod

```diff
@@ -4,13 +4,15 @@
 
 module github.com/upbound/provider-azuread
 
-go 1.21
+go 1.21.3
```

Comment:
It seems that the reason for this patch version pin is [here](https://github.com/upbound/terraform-provider-azuread/blob/ce1f46c210208225440f9297be387d2102c9b136/go.mod#L65).